pkg/util/miseadapter/authorizer.go

```diff
@@ -0,0 +1,74 @@
+package miseadapter
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type MISEAdapter interface {
+	IsAuthorized(ctx context.Context, r *http.Request) (bool, error)
+	IsReady() bool
+}
+
+type miseAdapter struct {
+	client *Client
+	log    *logrus.Entry
+}
+
+func NewAuthorizer(miseAddress string, log *logrus.Entry) MISEAdapter {
+	return &miseAdapter{
+		client: New(http.DefaultClient, miseAddress),
+		log:    log,
+	}
+}
+
+func (m *miseAdapter) IsAuthorized(ctx context.Context, r *http.Request) (bool, error) {
+	remoteAddr, _, _ := strings.Cut(r.RemoteAddr, ":")
+
+	i := Input{
+		OriginalUri:         fmt.Sprintf("http://%s%s", r.Host, r.URL.Path),
+		OriginalMethod:      r.Method,
+		OriginalIPAddress:   remoteAddr,
+		AuthorizationHeader: r.Header.Get("Authorization"),
+	}
+
+	result, err := m.client.ValidateRequest(ctx, i, m.log)
+	if err != nil {
+		return false, err
+	}
+
+	if result.StatusCode != http.StatusOK {
+		m.log.Errorf("mise authentication failed with %d: %s", result.StatusCode, result.ErrorDescription)
+		return false, nil
```

Comment:
Why we are return nil not the error?

Reply:
Two reasons. One is that we are already logging the error before returning and another is that we want this to go ahead for authentication via TLS method even if MISE authentication fails. We might investigate on the failure later and segregate it as valid or invalid.